developer-experience/readme.md

```diff
@@ -0,0 +1,130 @@
+# Developer Experience
+
+Developer experience refers to the how easy or difficult it is for a developer to perform essential tasks needed to implement a change. A positive developer experience would mean these tasks are relatively easy for the team (see measures below).
+
+The essential tasks are identified below.
+
+Build - Verify that changes are free of syntax error and compile.
+Test - Verify that all automated tests pass.
+Start - Launch end-to-end to simulate execution in a deployed environment.
+Debug - Attach debugger to started solution, set breakpoints, step through code, and inspect variables.
+
+If effort is invested to make these activities as easy as possible, the returns on that effort will be multiply the longer the project runs and the larger the team.
+
+## Goals
+
+- Maximize the amount of time engineers spend on writing code that fulfills story acceptance and done-done criteria.
+- Minimize the amount of time spent manual setup and configuration of tooling
+- Minimize regressions and new defects by making end-to-end testing easy
+
+## Impact
+
+Developer experience can have a significant impact on the efficiency of the day to day execution of the team. A positive experience can pay dividends throughout the lifetime of the project; especially as new developers join the team.
+
+- Increased Velocity - Team spends less time on non-value-add activities such as dev/local environment setup, waiting on remote environments to test, and rework (fixing defects).
+- Improved Quality - When its easy to debug and test, developers will do more of it. This will translate to fewer defects being introduced.
+- Easier Onboarding & Adoption - When dev essential tasks are automated, there is less documentation to write and, subsequently, less to read to get started!
+
+## Measures
+
+### Time to First Debug (aka F5 Contract)
+
+Assuming a laptop/pc that has never run the solution, how long does it take to start debugging the solution end-to-end.

```

Comment:
Nit: "end to end" is a bit of an overloaded terms for projects that fit into (or will fit into) a larger ecosystem of software. I think in this case you intend "end to end" to mean running the software that is being implemented as part of the engagement, and it might be worth it to rephrase to indicate that, or define what "end to end" means here.

Reply:
Good point. Agree this leaves much room for interpretation. I added a section at the start to define what it means in the context of this doc..

> ## Defining End-to-End
> 
> This document makes several references to running a solution end-to-end (aka E2E). End-to-end for the purposes of this document is scoped to the software that is owned, built, and shipped by the team. Systems owned by other teams or third-party vendors is not within the E2E scope for the purposes of this document.